### PR TITLE
Use okhttp and add DOH for didweb

### LINKED ISSUE #111: Use `okhttp` instead of `ktor`

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
Fixes #111 by using okhttp as the ktor engine. 

# Description
This also adds a client with DNS over HTTP for DIDWeb. 

# How Has This Been Tested?
Existing tests pass.

### CODE REVIEW

File: dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt
Comment (decentralgabe) [inline dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt:95]: why is this preferable? why choose 8.8.8.8 over 9.9.9.9 (quad9) or 1.1.1.1 (cloudflare)?

**Comment by phoebe-lew** [comment]
+1 to using okhttp as it's what we're using elsewhere, @andresuribe87 can you summarize why DNS over HTTP is better?

**Comment by mistermoe** [comment]
@andresuribe87 what are the benefits of using `ktor`'s `okhttp` adapter over vanilla `okhttp`?

**Comment by andresuribe87** [comment]
> +1 to using okhttp as it's what we're using elsewhere, @andresuribe87 can you summarize why DNS over HTTP is better?

Good question @phoebe-lew; First, a correction that I missed: it's DNS over HTTPS (not over HTTP). It makes it so that DNS resolution happens in a secure and privacy preserving manner. Thus, preventing man in the middle attacks. See https://w3c-ccg.github.io/did-method-web/#dns-security-considerations for details.

**Comment by andresuribe87** [comment]
> @andresuribe87 what are the benefits of using `ktor`'s `okhttp` adapter over vanilla `okhttp`?

@mistermoe good question. By using `ktor`, we allow users of the SDK to use whatever httpclient engine they use within their stack. While many wont touch it, other companies may want to have a custom, internal, client. Additionally, it keeps the door open to doing [multiplatform development](https://kotlinlang.org/docs/multiplatform-get-started.html). For example, if we support [Kotlin native](https://kotlinlang.org/docs/native-overview.html#why-kotlin-native) in this SDK, that could be another option for having a single codebase for Android, JVM, Linux, and iOS.

File: dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt
Comment (andresuribe87) [reply dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt:95]: This wasn't intentionally chosen. I copy pasted the setup as shown in https://github.com/square/okhttp/tree/master/okhttp-dnsoverhttps

Do you have any recommendations?

**Comment by andresuribe87** [comment]
Since CI was failing because of code coverage, I took to opportunity to set up integration testing. The configuration I followed tracked https://docs.gradle.org/current/userguide/java_testing.html#sec:configuring_java_integration_tests

I added a couple of very basic integration tests as well. 

Thus, this PR will also fix #57

File: dids/src/intTest/kotlin/web5/sdk/dids/DidIntegrationTest.kt
Comment (mistermoe) [inline dids/src/intTest/kotlin/web5/sdk/dids/DidIntegrationTest.kt:1]: can we have our integration test directory structure match the unit test directory structure?

**Comment by mistermoe** [comment]
i'm guessing we'll keep #111  open until we replace the `CIO` engine with `okhttp` across the entire project?

File: dids/src/intTest/kotlin/web5/sdk/dids/DidIntegrationTest.kt
Comment (andresuribe87) [reply dids/src/intTest/kotlin/web5/sdk/dids/DidIntegrationTest.kt:1]: I intentionally didn't do that because I expected integration tests to span across multiple packages and directories. For instance, I expect to add a test that reuses the same client across all different didmethods. I'm waiting until #118  is merged in order to do that. 

Given the above, what would you suggest?

File: dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt
Comment (decentralgabe) [reply dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt:95]: I would recommend quad9 for its strong privacy characteristics

**Comment by andresuribe87** [comment]
> i'm guessing we'll keep #111 open until we replace the `CIO` engine with `okhttp` across the entire project?

@mistermoe yes, that's the idea. I double checked that I had removed all instances that currently exist in the project, but probably missed some. Are there instances you found that I can remove?

File: dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt
Comment (andresuribe87) [reply dids/src/main/kotlin/web5/sdk/dids/DidWeb.kt:95]: Done in [9459174](https://github.com/TBD54566975/web5-kt/pull/140/commits/9459174f3c1b317af0210ef301429c45e5c2e998)

**Review by decentralgabe — APPROVED** [review]
lgtm

File: docs/examples.md
Comment (decentralgabe) [inline docs/examples.md:31]: any reason to change from 5 to 4?

Comment (andresuribe87) [reply docs/examples.md]: Not really, just an example :)

**Review by mistermoe — APPROVED** [review]

**Pull request merged**
